tests/haskell-utils-tests.el

```diff
@@ -225,4 +225,52 @@ removed."
            " TestTest Test test"
            (haskell-utils-reduce-string " Test\r\nTest\n    Test test"))))
 
+(ert-deftest post-command-hooks ()
+  "Test commands related `haskell-utils-async-post-command-flag'.
+Tests flag updates and `post-command-hook' cleanup."
+  (with-temp-buffer
```

Comment:
Use:

``el
(with-temp-switch-to-buffer ...
``

Note that `with-temp-switch-to-buffer` is currently defined in `haskell-indentation-tests.el`, it should be moved to the common file for test utilities we have been talking about.